File: Chapter09/HelpDeskCS/HelpDeskCS/HelpDeskCS.DesktopClient/Screens/CreateNewIssue.lsml.cs
// You can use and redistribute the code from this book (and sample application) for non-commercial and 
// most commercial purposes as long as you acknowledge the source and authorship. 
// You should note the source of the code in any documentation as well as in the program code itself (as a comment). 
// The acknowledgment should include author, title, publisher, and ISBN. 
// An example of such an acknowledgment would be "Derived from Listing 2-10, LightSwitch 2015 by Tim Leung, published by Apress, ISBN 978-1-4842-0767-3".

using Microsoft.LightSwitch.Presentation.Extensions;
using Microsoft.LightSwitch.Presentation;
using Microsoft.LightSwitch.Framework.Client;
using Microsoft.LightSwitch;
using System.Collections.Generic;
using System.IO.IsolatedStorage;
using System.IO;
using System.Linq;
using System;

namespace LightSwitchApplication
{
    public partial class CreateNewIssue
    {
        partial void CreateNewIssue_InitializeDataWorkspace(global::System.Collections.Generic.List<global::Microsoft.LightSwitch.IDataService> saveChangesTo)
        {
            this.IssueProperty = new Issue();


            //Listing 9 - 3.Setting default control values
            this.IssueProperty = new Issue();
            this.IssueProperty.TargetEndDateTime = DateTime.Now.AddDays(3);
            this.IssueProperty.Priority = PriorityMedium;


            //Listing 9 - 10.Setting the focus to a control 
            this.FindControl("ProblemDescription").Focus();


            if (EngineerIDParam.HasValue)
            {

                //Listing 9 - 7.Using screen parameter values in code
                this.IssueProperty.Engineer =
                 this.DataWorkspace.ApplicationData.Engineers_SingleOrDefault(
                   EngineerIDParam.Value);

            }

        }

        partial void CreateNewIssue_Saved()
        {
            //this.Close(false);
            //Application.Current.ShowDefaultScreen(this.IssueProperty);


            //Listing 9 - 18.Reset the New Data Screen after a save
            this.IssueProperty = new Issue();

        }
    }
}

File: Chapter09/HelpDeskVB/HelpDeskVB/HelpDeskVB.DesktopClient/Screens/CreateNewIssue.lsml.vb
'You can use and redistribute the code from this book (and sample application) for non-commercial and 
'most commercial purposes as long as you acknowledge the source and authorship. 
'You should note the source of the code in any documentation as well as in the program code itself (as a comment). 
'The acknowledgment should include author, title, publisher, and ISBN. 
'An example of such an acknowledgment would be "Derived from Listing 2-10, LightSwitch 2015 by Tim Leung, published by Apress, ISBN 978-1-4842-0767-3".

Namespace LightSwitchApplication

    Public Class CreateNewIssue

        'Listing 9-3. Setting default control values

        Private Sub CreateNewIssue_InitializeDataWorkspace(saveChangesTo As List(Of IDataService))

            Me.IssueProperty = New Issue()
            Me.IssueProperty.TargetEndDateTime = DateTime.Now.AddDays(3)
            Me.IssueProperty.Priority = PriorityMedium


            'Listing 9 - 10.Setting the focus to a control 
            Me.FindControl("ProblemDescription").Focus()

            If EngineerIDParam.HasValue Then

                'Listing 9 - 7.Using screen parameter values in code
                Me.IssueProperty.AssignedEngineer =
                 Me.DataWorkspace.ApplicationData.Engineers_SingleOrDefault(
                   EngineerIDParam.Value)

            End If


        End Sub

        Private Sub CreateNewIssue_Saved()

            'Listing 9 - 18.Reset the New Data Screen after a save
            'Me.Close(False)
            'Application.Current.ShowDefaultScreen(Me.IssueProperty)
            Me.IssueProperty = New Issue

        End Sub

        Private Sub CreateNewIssue_Created()

        End Sub
    End Class

End Namespace